File: Coding Problems & My Solutions/Container With Most Water/Container_With_Most_Water.java
class Solution {
    public int maxArea(int[] height) {
        int max = 0;
        int a = 0;
        int b = height.length - 1;
        while (a < b){
            max = Math.max(max,(Math.min(height[a],height[b])*(b-a)));
            if (height[a] > height[b]){
                b--;
            } else{
                a++;
            }
        }
        return max;
    }
}

class Solution {
    public int maxArea(int[] height) {
        int max = 0;
        int a = 0;
        int b = height.length - 1;
        while (a < b) {
            int area = Math.min(height[a], height[b]) * (b - a);
            max = Math.max(max, area);
            if (height[a] > height[b]) {
                b--;
            } else {
                a++;
            }
        }
        return max;
    }
}

File: Coding Problems & My Solutions/Container With Most Water/Container_With_Most_Water.swift
class Solution {
    func maxArea(_ height: [Int]) -> Int {
        var maxNo = Int.min
        var a = 0
        var b = height.count - 1
        while a <= b {
            maxNo = max(maxNo, (b-a) * min(height[a], height[b]))
            if height[a] > height[b] {
                b -= 1
            } else {
                a += 1
            }
        }
        return maxNo
    }
}

class Solution {
    func maxArea(_ height: [Int]) -> Int {
        var a = 0
        var b = height.count - 1
        var ans = 0
        while a < b {
            let localArea = min(height[a], height[b]) * (b - a)
            ans = max(ans, localArea)
            if height[a] > height[b] {
                b -= 1
            } else {
                a += 1
            }
        }
        return ans
    }
}